server/tests/controllers/miners/github/test_release.py

```diff
@@ -1403,15 +1412,18 @@ async def test_precomputed_releases_ambiguous(
         ["src-d/go-git"], branches, default_branches, time_from, time_to,
         release_match_setting_branch, 1, (6366825,), mdb, pdb, rdb, None)
     await wait_deferred()
+    if with_preloading_enabled:
+        await pdb.cache.refresh()
     prels = await release_loader._fetch_precomputed_releases(
         {ReleaseMatch.tag: {".*": ["src-d/go-git"]},
          ReleaseMatch.branch: {"master": ["src-d/go-git"]}},
         time_from, time_to, 1, pdb)
-    prels = prels[releases_tag.columns]
+    cols = set(releases_tag.columns).intersection(prels.columns)
```

Comment:
The columns should be exactly the same, hence the check. If they don't then there is a bug.

Reply:
When I was testing the releases table everything worked so when building the options for the `MemoryCache` I just skipped all the columns that were unused. The reasons being to save memory ofc, especially given that many of the columns skipped are identifiers. Or maybe I just missed where they're used? At least for the unfresh path it didn't break.
What I mean is that we can easily just preload all the columns, but it will be costly. Does it make sense to change this check only when preloading is enabled then?

Reply:
As I remember, `_fetch_precomputed_releases` should return quite specific columns that much the "live-loaded" alternative. The code really relies on all of them. What are the exact unused columns?

Reply:
The only columns that are being preloaded are the followings:
```
                    PrecomputedRelease.id,
                    PrecomputedRelease.release_match,
                    PrecomputedRelease.repository_full_name,
                    PrecomputedRelease.published_at,
                    PrecomputedRelease.acc_id,
```

But probably I know why: https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/controllers/features/github/unfresh_pull_request_metrics.py#L93

Here `releases` should be the processed data frame that comes from `_fetch_precomputed_releases`. But inside `load_releases` the columns that I mentioned above are enough, and nothing else is needed. What I'm saying is based on how the `releases` variable [here](https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/controllers/miners/github/release_load.py#L103) is used inside the method.

Reply:
Changed conditionally on `with_preloading_enabled`.

Reply:
This is the thing. There are code branches that require the rest of the columns. For example, filtering PRs breaks here: https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/controllers/features/github/pull_request_filter.py#L256
Or filtering JIRA and `/get/pull_requests` break here: https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/controllers/features/github/pull_request_filter.py#L787
The tests do not break because the property values are not checked, probably...

Reply:
So we should preload everything, really.

Reply:
I understood: we are only running "fat" tests so some controller tests do not execute with preloading.

Reply:
Let's do the following then. Explicitly assert that the rest of the columns do not exist under `with-preloading`. This way, when we extend to the other endpoints, these tests will fail and make us update them. Add a comment that explains why we assert and how to update - to remove the `if` whatsoever.

Reply:
It makes a lot of sense, thanks for checking this out 🙇